iavl/iavl_node.go

```diff
@@ -522,5 +552,31 @@ func removeOrphan(t *IAVLTree, node *IAVLNode) {
 	if t.ndb == nil {
 		return
 	}
+
+	// Don't orphan persistent nodes if another tree depends on it
+	if Committed != nil && t != Committed {
+		tree := Committed
+		exists := tree.root.findNode(tree, node)
+		if exists {
+			return
+		}
+	}
+
+	if DeliverTx != nil && t != DeliverTx {
```

Comment:
Yea, I realized afterwards that we don't even need to 'name' the trees, it can just be a list of them that is built up by the Copy function and reset by Save. That generalizes it a bit more, but...